src/libraries/System.Net.Security/src/System/Net/Security/SslStreamPal.Windows.cs

```diff
@@ -332,19 +333,20 @@ public static unsafe SecurityStatusPal DecryptMessage(SafeDeleteSslContext? secu
 
                 // Decrypt may repopulate the sec buffers, likely with header + data + trailer + empty.
                 // We need to find the data.
-                count = 0;
+                outputCount = 0;
                 for (int i = 0; i < NumSecBuffers; i++)
                 {
                     // Successfully decoded data and placed it at the following position in the buffer,
                     if ((errorCode == Interop.SECURITY_STATUS.OK && unmanagedBuffer[i].BufferType == SecurityBufferType.SECBUFFER_DATA)
                         // or we failed to decode the data, here is the encoded data.
                         || (errorCode != Interop.SECURITY_STATUS.OK && unmanagedBuffer[i].BufferType == SecurityBufferType.SECBUFFER_EXTRA))
                     {
-                        offset = (int)((byte*)unmanagedBuffer[i].pvBuffer - bufferPtr);
-                        count = unmanagedBuffer[i].cbBuffer;
+                        outputOffset = (int)((byte*)unmanagedBuffer[i].pvBuffer - bufferPtr);
+                        outputCount = unmanagedBuffer[i].cbBuffer;
 
-                        Debug.Assert(offset >= 0 && count >= 0, $"Expected offset and count greater than 0, got {offset} and {count}");
-                        Debug.Assert(checked(offset + count) <= buffer.Length, $"Expected offset+count <= buffer.Length, got {offset}+{count}>={buffer.Length}");
+                        // output is ignored on Windows. We always decrypt in place and we set outputOffset to indicate where the data start.
+                        Debug.Assert(outputOffset >= 0 && outputCount >= 0, $"Expected offset and count greater than 0, got {outputOffset} and {outputCount}");
```

Comment:
I still think it's a bad idea to have the PAL work differently on Windows and Linux. I.e., we now have an `output` param that is used in the Linux version of this method, but not in the Windows one.

Unless we can figure out how to define the PAL in such a way that it works the same way on all platforms, can we just leave the signature the way it is? Or at least do minimal Span-ification? Meaning, pass `Span<byte> buffer` and change `offset` and `count` to be out params instead of ref.

Reply:
I could make it same. However on Windows this would take another copy. I can measure the impact but I did not want to do that. One way how to deal with it could be pushing more functionality to PAL.

I had several different versions and I felt this is best option. It may go ahead little bit as it may make more sense when we use user provided buffer directly. The way how I think about it now the `Output` is hint where we want data to land. Then the PAL tells us back if it did or did not. There may be better semantic how to do that.  I can also add more commentary with explanation.

Reply:
Why is it another copy on Windows?

I'm basically saying, don't modify the PAL at all right now; leave it as is. The only change worth considering as part of this PR would seem to be to use Span, but we can do that without changing the semantics of the function.

Reply:
I can mock up what I mean in code if that would be helpful.

Reply:
yes, I think most would be nice. I can get that in next pass.

Reply:
Ok, I'll ignore this and related changes for now.

Reply:
I roll-back the input/output split and I'll do it is separate change.